source/lba1/compile.rst

```diff
@@ -1,2 +1,143 @@
 Compile
-=======
\ No newline at end of file
+=======
+
+Prerequisites
+-------------
+
+-  `DOSBox <https://www.dosbox.com/>`__ - DOS emulator which we will
+   use to compile the game inside.
+-  `4DOS <https://www.4dos.info/v4dos.htm#751>`__ - Command line
+   interpreter, which supports the ``copy`` command with binary
+   inputs and output.
+-  Watcom 10 compiler - For compiling C sources and running MAKEFILEs
+-  MASM (Microsoft Macro Assembler) 6.0 - For compiling ASM sources
+
+Getting prerequisites and sources
+---------------------------------
+
+DOSBox and 4DOS are freely available. For getting Watcom 10 and MASM
+6.0, you need to search the internet. Note that we did not manage to
+build the game with Open Watcom. Also, for some reason the MASM version
+6.11 compiler did run very slowly in the DOSBox, so it was basically
+unusable. We had to use the version 6.0.
+
+All directories and files will placed in the ``~/lba-hacking`` directory
+on the host machine. Feel free to change this path, but then adjust the
+DOSBox configuration below correspondingly. This directory will be
+mounted to ``C:`` in DOSBox.
+
+-  Extract 4DOS into ``4dos``.
+-  Extract Watcom and MASM installers into ``install``. These will be
+   needed to be installed.
+-  Clone https://github.com/2point21/lba1-classic-community into
+   ``lba``.
+
+The dir structure at this point should like something like this:
+
+::
+
+   ~/lba-hacking
+   ├── 4dos
+   ├── install
+   │   ├── masm
+   │   └── watcom
+   └── lba
+
+DOSBox configuration
+--------------------
+
+Change the ``autoexec`` section of you DOSBox configuration like below.
+The configuration path of DOSBox is usually shown when you start it.
+
+::
+
+   [autoexec]
+   mount C ~/lba-hacking
+
+   PATH c:\watcom\binw;c:\masm\bin;%PATH%
+   set INCLUDE=c:\watcom\h;c:\lba\lib386
```

Comment:
To compile LIB_MIDI and LIB_SAMP, an additional include  `c:\ail2` was necessary with the contents in this [repository] (https://github.com/Tronix286/AIL2) . This doesn't seem to be the AIL version used originally in LBA, which could result in some incompatibilities, but so far the build appears to be working.